lib/Sema/TypeCheckDecl.cpp

```diff
@@ -6623,18 +6623,26 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       
       // Make sure that the overriding property doesn't have storage.
       if (overrideASD->hasStorage() && !overrideASD->hasObservers()) {
-        auto diagID = diag::override_with_stored_property;
+        bool downgradeToWarning = false;
         if (!TC.Context.isSwiftVersionAtLeast(5) &&
             overrideASD->getAttrs().hasAttribute<LazyAttr>()) {
           // Swift 4.0 had a bug where lazy properties were considered
           // computed by the time of this check. Downgrade this diagnostic to
           // a warning.
-          diagID = diag::override_with_stored_property_warn;
+          downgradeToWarning = true;
+          // HACK: We would normally do this at the end of validation, but
+          // if we wait that long then the getter doesn't get marked as an
+          // override.
+          maybeAddAccessorsToVariable(cast<VarDecl>(overrideASD), TC);
```

Comment:
Trivial getters are created in CodeSynthesis.cpp, maybe we should mark them as overrides there if the AbstractStorageDecl overrides something? Up until now this could not happen because trivial accessors are only created for stored properties and lazy properties which normally don't override anything.

Reply:
Oh, that might make more sense. I could see `maybeAddAccessorsToVariable` depending on something earlier in validation.